src/main/scala/rise/elevate/rules/package.scala

```diff
@@ -16,7 +16,7 @@ package object rules {
     case App(Lambda(x, b), v) =>
       Success(substitute.exprInExpr(v, `for` = x, in = b))
     case DepApp(k1, DepLambda(k2, x, b), v) if k1 == k2 =>
-      Success(substitute.kindInExpr(k1, v, `for` = x, in = b))
+      Success(substitute.kindInExpr(k2, v, `for` = x, in = b))
```

Comment:
this shouldn't change anything (as `k1 == k2`). why the change?

Reply:
This is because Scala's type system has not internalised the fact that `k1 == k2`, and thus `x` is of the type `I` bound in `k2`, not the one bound in `k1`.